examples/Imageflow.Server.Example/Startup.cs

```diff
@@ -31,10 +32,40 @@ public Startup(IConfiguration configuration, IWebHostEnvironment env)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllersWithViews();
-            
-            services.AddImageflowRemoteReaderService(new RemoteReaderServiceOptions { SigningKey = "ChangeMe"}
-                .AddPrefix("/remote/")
-                );
+
+            services.AddHttpClient("TrickyHttpClient", config =>
+            {
+                config.DefaultRequestHeaders.Add("user-agent", "Tricky Client");
+                config.DefaultRequestHeaders.Add("authorization", "secret");
+            });
+
+            services.AddHttpClient(nameof(RemoteReaderService), config =>
+            {
+                config.DefaultRequestHeaders.Add("user-agent", "ImageFlow-DotNet-Server");
+
+            }).ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
+            {
+                AllowAutoRedirect = true,
+                MaxAutomaticRedirections = 50
+
+            }).AddTransientHttpErrorPolicy(builder => builder
+                .RetryAsync()
+            );
+
+            var remoteReaderServiceOptions = new RemoteReaderServiceOptions
+            {
+                SigningKey = "ChangeMe"
+            }
+            .AddPrefix("/remote/");
+
+            services.AddImageflowRemoteReaderService(remoteReaderServiceOptions, uri =>
```

Comment:
Shouldn't the selector be a Func<string> property on RemoteReaderServiceOptions instead of having an overload?

Reply:
I can move to the options.  Are you doing anything like this elsewhere that I can refer to for consistency?